Configuring Shadcn to work with Vite/React and creating ToasterWithTitle component

Stevensauro opened the pull request:
Any notes are appreciated!!

Stevensauro committed:
setting up shadcn for toaster component

Stevensauro committed:
Configuring Shadcn to work with Vite and created ToasterWithTitle component

Stevensauro committed:
Add shadcn accordion and update tabSwitcher component mobile design

Stevensauro committed:
Created Heart & HeartIcon  components.

- Update hearted elements to local storage
- if the element was hearted the color will be filled up on the icon
- Elements Ids will be saved in local storage to reference them later maybe in the fav items container ? or also to grab data from elements like in the color picker for the hex values etc.

Stevensauro committed:
Update Heart Component and clean a little bit more

joekotvas commented on fetools-app/src/components/Heart.jsx:
```diff
@@ -0,0 +1,64 @@
+        const favoriteItems = JSON.parse(favoriteItemsJSON)
+
+        if(!favoriteItems.includes(elementId)){
+            favoriteItems.push(elementId)
```
This component is great. We do need to allow it to be a bit more flexible, so instead of accepting an elementId, lets allow it to accept favoriteContent.

For the character finder, it will simply be a character. For the color picker, it will be a hex color code. For the font visualizer, it will be an object of font settings.

joekotvas reviewed (approved):
Looks good! I added a note on the favorite button functionality for further development.

One other minor detail; as the toasts are in the bottom center of the screen in our design, they need to exit downward rather than to the right.

Stevensauro commented:
> Looks good! I added a note on the favorite button functionality for further development.
> 
> One other minor detail; as the toasts are in the bottom center of the screen in our design, they need to exit downward rather than to the right.

Haha that's what I wanted to ask you today 🤣. I'm not too sure how to fix that with shadcn, I changed some of the style but I couldn't figure out how to make it go downwards or upwards
